### ISSUE TITLE
shared mongodb with multiple masters that only watch for certain jobs?

### ISSUE THREAD

**Issue by aon108** [nathanschwarz/meteor-cluster#8]
Is there a way to configure the Master server to watch only specific jobs on a shared mongodb queue from multiple apps so that you can have multiple master servers doing different things?

**Comment by karthage-spartacus** [comment]
right now no, but it's something that will be introduced in later versions.
I'll try to update when I can.

I gave an advise about this use case in the README : you should have 1 app dedicated to handling all the tasks.

**Comment by aon108** [comment]
I did check out all of the README and was thinking that I could just define a unique taskmap for each Master and they wont try run each others jobs but looking at the code I think this change might get the functionality working, what do you think?

- In the taskQueue.js where it pulls the jobs from the mongodb [this.pull](https://github.com/nathanschwarz/meteor-cluster/blob/1af0e3bbefcd816e18058f7afb88e90af1fc9176/src/TaskQueue.js#L79)
- adding the mongodb query filter to only return the tasks that are found inside the taskMap
```
const taskTypes = _.pluck(this.taskMap,"taskType");
return this.rawCollection().findOneAndUpdate({ taskType: taskTypes, onGoing: false, dueDate: { $lte: new Date() }}, { $set: { onGoing: true }}, { sort: { priority: -1, createdAt: 1, dueDate: 1 }})
```
and of course update the jobs counts with the same taskType filter [line 93](https://github.com/nathanschwarz/meteor-cluster/blob/1af0e3bbefcd816e18058f7afb88e90af1fc9176/src/TaskQueue.js#L93)

**Comment by karthage-spartacus** [comment]
Yes that's exactly what I had in mind ☺️ but I'd like to avoid underscorejs
/ lodash, so just use `Object.keys` instead of _.pluck.

If you want you can make a pull request :)
